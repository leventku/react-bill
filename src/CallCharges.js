import React from 'react';

import BillGroup from './BillGroup';

function CallCharges(props) {
  const renderItem = (item, i) => (
    <li className="list-group-item justify-content-between" key={i}>
      <span className="number" style={{width: 100}}>{item.called}</span>
      <span className="duration">{item.duration}</span>
      <span className="cost">Â£{item.cost.toFixed(2)}</span>
    </li>
  );

  return (
    <BillGroup
      title="Call Charges"
      items={props.data.calls}
      itemRenderer={renderItem}
      total={props.data.total}
    />
  )
}

export default CallCharges;
