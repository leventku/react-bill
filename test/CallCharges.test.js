import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CallCharges from '../src/CallCharges';

function setupComponent() {
  const props = {
    data: {
      calls: [
        { called: '1', duration: '2', cost: 3 },
        { called: '2', duration: '2', cost: 3 },
        { called: '3', duration: '2', cost: 3 },
      ],
    },
  };

  const wrapper = shallow(<CallCharges {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('CallCharges component', () => {
  it('should render', () => {
    const { wrapper } = setupComponent();
    expect(wrapper.exists()).toBe(true);
  });

  it('should render its items', () => {
    const { wrapper } = setupComponent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
