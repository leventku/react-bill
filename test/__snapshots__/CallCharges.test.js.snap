// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallCharges component should render its items 1`] = `
<BillGroup
  itemRenderer={[Function]}
  items={
    Array [
      Object {
        "called": "1",
        "cost": 3,
        "duration": "2",
      },
      Object {
        "called": "2",
        "cost": 3,
        "duration": "2",
      },
      Object {
        "called": "3",
        "cost": 3,
        "duration": "2",
      },
    ]
  }
  title="Call Charges"
/>
`;
